modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/JsonProcessor.java

```diff
@@ -78,7 +78,7 @@ public void execute(IngestDocument document) throws Exception {
         Object fieldValue = document.getFieldValue(field, Object.class);
         BytesReference bytesRef = (fieldValue == null) ? new BytesArray("null") : new BytesArray(fieldValue.toString());
         try (XContentParser parser = JsonXContent.jsonXContent
-                .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, bytesRef)) {
+                .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, bytesRef.streamInput())) {
```

Comment:
These input streams need to be closed.

Reply:
I believe closing the XContentParser closes the internal JsonParser, which closes the stream.

Reply:
That’s correct but it’s not enough. An exception could be thrown constructing the parser in which case the stream will not be closed.